pydantic_settings/main.py

```diff
@@ -412,6 +422,7 @@ def _settings_build_values(
         env_file_encoding=None,
         env_ignore_empty=False,
         env_nested_delimiter=None,
+        env_nested_depth=-1,
```

Comment:
Can't we user default value of `None`?

Reply:
-1 is the default value of `maxsplit` in `str.split`. We can use `None` but we have to map it to -1 before passing it to `split`.